intellij-ext/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -0,0 +1,38 @@
+package com.metalbear.mirrord
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.actionSystem.ex.ComboBoxAction
+import javax.swing.JComponent
+
+class MirrordConfigDropDown : ComboBoxAction() {
+    private lateinit var configPaths: ArrayList<String>
+    companion object {
+        var selectedConfig = "Select Configuration"
+    }
+
+    override fun createPopupActionGroup(button: JComponent?): DefaultActionGroup {
+        val actionGroup = DefaultActionGroup()
+        configPaths.forEach { configPath ->
+            val trimmedPath = configPath.split("/").takeLast(3).joinToString("/")
+            actionGroup.add(object : AnAction(configPath) {
+                override fun actionPerformed(e: AnActionEvent) {
+                    selectedConfig = trimmedPath
+                    e.presentation.text = selectedConfig
+                }
+            })
+        }
+        return actionGroup
+    }
+
+    override fun update(e: AnActionEvent) {
+        val project = e.project ?: throw Error("required to have configured project to use mirrord config.")
+        configPaths = MirrordConfigAPI.searchConfigPaths(project) as ArrayList<String>
```

Comment:
this isn't good; update can be called every half second which means it might add a lot of overhead to call this.
I think the way to go is to register a BulkFileListener that updates the available files and then use it's value.
Please notice that `BulkFileListener` isn't project level so need to filter out files not in the project.

Reply:
okay so, should we initialize the configs in the constructor and then listen for events, so when a file that is a config file i.e. mirrord.json is moved anywhere/copied/deleted we update our list?

Reply:
yes